layer_metadata_restorer.go

```diff
@@ -45,22 +50,60 @@ func (la *DefaultLayerMetadataRestorer) Restore(buildpacks []buildpack.GroupBuil
 func (la *DefaultLayerMetadataRestorer) restoreStoreTOML(appMeta platform.LayersMetadata, buildpacks []buildpack.GroupBuildpack) error {
 	for _, bp := range buildpacks {
 		if store := appMeta.MetadataForBuildpack(bp.ID).Store; store != nil {
-			if err := WriteTOML(filepath.Join(la.LayersDir, launch.EscapeID(bp.ID), "store.toml"), store); err != nil {
+			if err := WriteTOML(filepath.Join(la.layersDir, launch.EscapeID(bp.ID), "store.toml"), store); err != nil {
 				return err
 			}
 		}
 	}
 	return nil
 }
 
+type layerToSha struct {
+	layerToShaMap map[string]string
+}
+
+func initializeLayerToSha() layerToSha {
```

Comment:
`initializeLayerToSha` is being called `addLayerToMap`, and it may happen more than once.
I think you wanted to put your comment on `initializeBuildpackLayersToSha`.
I don't have a strong opinion on it but I think it's clearer this way, especially if it's part of `restoreLayerMetadata` (and not as part of `NewLayerMetadataRestorer` as you suggested [here](https://github.com/buildpacks/lifecycle/pull/624#discussion_r643450316)).

Reply:
Does it happen more than once? I missed that. Personally I think having the separate function makes the code harder to read.

Reply:
It happens for every `buildpackID` that is being added to the map.
`restoreLayerMetadata` is looping over `buildpacks`, then inside we're looping over the layers and calling to `writeLayerMetadata` when needed. There we call `addLayerToMap` that will call `initializeLayerToSha` for the first layer of a `buildpackID`.
I think that if you'll review it through an IDE (instead of Github), it'll be easier for you to see it.
Personally, I think it's clearer to have it as a separate function.

Reply:
Ohh I just meant that `initializeLayerToSha` appears only twice in the code (where it's declared, and where it's invoked). But I am happy to keep it as-is if you feel strongly about it.

Reply:
yeah, both functions appear only twice in the code - where they declared and where they invoked.

Reply:
I changed this code as part of my last commit.